api/dataset.go

```diff
@@ -42,17 +43,48 @@ var (
 	}
 )
 
+/*
+   1. check the query parameter, if not exists... as expected  if exists
+   if is_based_on exists  and but value is empty string (no value given) then 400
+   2. if there, then query database for different
+   3. ensure that the swagger spec is respected
+
+*/
+
+const IsBasedOn = "is_based_on"
+
 //getDatasets returns a list of datasets, the total count of datasets and an error
 func (api *DatasetAPI) getDatasets(w http.ResponseWriter, r *http.Request, limit int, offset int) (interface{}, int, error) {
 	ctx := r.Context()
 	logData := log.Data{}
 	authorised := api.authenticate(r, logData)
-	datasets, totalCount, err := api.dataStore.Backend.GetDatasets(ctx, offset, limit, authorised)
+
+	isBasedOnExists := r.URL.Query().Has(IsBasedOn)
+	isBasedOn := r.URL.Query().Get(IsBasedOn)
+
+	if isBasedOnExists && isBasedOn == "" {
+		err := apierrors.ErrInvalidQueryParameter
+		log.Error(ctx, "malformed is_based_on parameter", err)
+		handleDatasetAPIErr(ctx, err, w, logData)
+		return nil, 0, err
+	}
+
+	var datasets []*models.DatasetUpdate
+	var totalCount int
+	var err error
+
+	if isBasedOnExists {
+		datasets, totalCount, err = api.dataStore.Backend.GetDatasetsByBasedOn(ctx, isBasedOn, offset, limit, authorised)
+	} else {
+		datasets, totalCount, err = api.
```

Comment:
It's probably better/more common  to format this line like this:

```
datasets, totalCount, err = api.dataStore.Backend.GetDatasets(
  ctx,
  offset,
  limit,
  authorised,
)
```
If you want to shorten the line length. Using multiple lines after each `.` is usually reserved for when multiple functions are being chained together e.g.
```
Then.Do().
Stuff().
ETC()
```